src/core/tensor/tensor_math_cpp.h

```diff
@@ -868,6 +868,25 @@ void RowMax<float, lang::Cpp>(const Tensor& in, Tensor *out, Context *ctx) {
   }
 }
 
+template <>
+void SoftMax<float, lang::Cpp>(const Tensor& in, Tensor* out, Context *ctx) {
```

Comment:
do we need to move SoftMax into tensor_math_cpp.h?
If not, please delete the changes.

Reply:
Hi @nudles , it is not implemented in cpp yet. So we need to move this from `tensor.cc` into `tensor_math_cpp.h`. If not, when calling softmax with cpp, it will through `not impl` fatal error.